### DISCUSSION TITLE
Add an option for `toFormData` to allow null values

### DISCUSSION THREAD

**Discussion by maclementED** [axios/axios#6329] [General]
I needed to send null values in a FormData. Sadly, there wasn't any way to indicate to keep null values as is. 

This [line](https://github.com/axios/axios/blob/release/lib/helpers/toFormData.js#L117) is causing to transform null to empty string. 
```
 function convertValue(value) {
    if (value === null) return '';
...

```
We would need to update `options` object and `convertValue` function to accept a parameter like `nullsAsEmptyString` 

If needed, I can submit a PR for this feature

**Comment by jasonsaayman** [comment]
`FormData` cannot preserve JavaScript `null` as a real `null` value on the wire. A multipart field is ultimately bytes plus headers, and `FormData.append()` accepts a string/Blob/File-like value. So Axios has to choose a representation, such as:

- omit the field
- send an empty string
- send the string `"null"`
- send JSON, for example `{"field": null}`

Changing the default behaviour would be risky because existing users may depend on the current null handling.

For now, the safest workaround is to choose an explicit representation before passing the data to Axios.

Example: send `"null"` as a string:

```js
function serializeNulls(value) {
  if (value === null) {
    return 'null';
  }

  if (Array.isArray(value)) {
    return value.map(serializeNulls);
  }

  if (value && typeof value === 'object') {
    return Object.fromEntries(
      Object.entries(value).map(([key, val]) => [key, serializeNulls(val)])
    );
  }

  return value;
}

await axios.postForm('/api', serializeNulls(data));
```

Or build the `FormData` manually when the backend requires a specific convention:

```js
const form = new FormData();

form.append('name', 'John');
form.append('deletedAt', 'null'); // backend must interpret this as null

await axios.post('/api', form);
```

If the backend supports JSON inside one multipart field, that is usually clearer:

```js
const form = new FormData();

form.append('payload', JSON.stringify({
  name: 'John',
  deletedAt: null
}));

await axios.post('/api', form);
```

A PR could make sense, but the option should define the exact wire behaviour. For example, something like `nulls: 'omit' | 'empty' | 'string'` would be clearer than “allow null values”, because raw `null` is not a multipart/form-data value.

Also note that in the current Axios code, `null` values are filtered before the custom `visitor` is called in the normal object traversal path, so `formSerializer.visitor` alone is not enough to preserve nulls. A PR would need tests for object properties, nested properties, and flat arrays.
